src/main/java/com/openlattice/shuttle/dates/JavaDateTimeHelper.java

```diff
@@ -1,81 +1,123 @@
             try {
                 DateTimeFormatter formatter = Cached.getDateFormatForString( datePattern );
                 R result = parseFunction.apply( date, formatter );
                 return postParseFunction.apply( result, datePattern );
-            } catch ( DateTimeParseException e) {
-                if ( datePattern.equals(datePatterns[datePatterns.length - 1]) ) {
-                    logger.error("Unable to parse date {}, please see debug log for additional information: {}.", date,e);
+            } catch ( DateTimeParseException e ) {
+                if ( datePattern.equals( datePatterns[ datePatterns.length - 1 ] ) ) {
+                    logger.error( "Unable to parse date {}, please see debug log for additional information: {}.",
+                            date,
+                            e );
                 }
             } catch ( ExecutionException ex ) {
-                logger.error("ExecutionException loading pattern from cache", ex);
+                logger.error( "ExecutionException loading pattern from cache", ex );
             }
+
         }
         return null;
     }
 
-    public LocalDate parseDate(String date) {
+    public LocalDate parseDate( String date ) {
         return parseWithTwoDigitYearHandling( date, LocalDate::parse, ( ld, datePattern ) -> {
-            if (  checkDatePatternIsTwoDigitYear( datePattern ) ) {
+            if ( checkDatePatternIsTwoDigitYear( datePattern ) ) {
                 // TODO: break this out into its own transform that specifies the date boundaries for two-digit years
-                if ((ld.getYear() - LocalDate.now().getYear()) > 20) {
-                    ld = ld.withYear(ld.getYear() - 100);
+                if ( ( ld.getYear() - LocalDate.now().getYear() ) > 20 ) {
+                    ld = ld.withYear( ld.getYear() - 100 );
                 }
             }
             return ld;
-        });
+        } );
     }
 
-    public OffsetDateTime parseDateTime(String date) {
-        return parseWithTwoDigitYearHandling( date, ( toParse, formatter ) -> {
-            LocalDateTime parsed = LocalDateTime.parse( toParse, formatter );
-            return parsed.atZone( tz.toZoneId() ).toOffsetDateTime();
-        }, ( odt, datePattern ) -> {
-            if (  checkDatePatternIsTwoDigitYear( datePattern ) ) {
-                // TODO: break this out into its own transform that specifies the date boundaries for two-digit years
-                if ((odt.getYear() - LocalDate.now().getYear()) > 20) {
-                    odt = odt.withYear(odt.getYear() - 100);
-                }
+    public OffsetDateTime parseDateTime( String date ) {
+        try {
+            // DateTime: first try parsing as iso-string into OffsetDateTime
+            // TimeZone: if Timezone is not consistent with reported TimeZone: warning
+            OffsetDateTime odt = OffsetDateTime.parse( date );
+            if ( tz.isPresent() ) {
```

Comment:
These checks seem to be copy pasted throughout the file -- I'd recommend creating a helper for this, i.e.
```
private void checkTimezonesMatch( OffsetDateTime odt ) {
  tz.ifPresent( timezone -> {
    if ( timezone.toZoneId() == odt.getOffset() ) {
      logger.error( "The reported and requested timezones are inconsistent." );
    }
  } );
}
```

Another question -- this is logging an error if the zoneId == the odt offset. Should that be the other way around?

Also you may want to consider throwing an exception instead of just logging an error. Seems like if we run into this case something has gone very wrong, and we might want to prevent the integration from running unless it's fixed.

Reply:
Yes and done on both comments.

For the third, we discussed with the data team what we want to do with this (including voting).

Having this throw an error will break a lot of the existing integrations,  we'd prefer a warning where we can at least in the beginning look at the logs and address the warnings to prevent the full integration from breaking.
Also, this warning will also be thrown if we parse a UTC-datestring, which might have been correct given that postgres doesn't store timezones.  So in that case it would be a false log.

Reply:
(another thing -- since ZoneId isn't a primitive you'll want to use `if ( !timezone.toZoneId().equals( odt.getOffset() )` )

Reply:
this is now fixed with passing tzId everywhere.